### PR TITLE
SUP-1424, New member registration for challenges. Fix from community.

### PR DESCRIPTION
**Pull request by vikasrohit**
@parthshah @nlitwin fyi..
These are the changes from community plus some my changes to allow reproduction of the issue in the vagrant env.

### CODE REVIEW

File: src/js/terms.js [hunk 1]
```diff
@@ -49,13 +49,13 @@ appChallengeTerms = {
               $('.agree-label').hide();
               $('#termSubmit').text('Go Back');
               $('.agreement').removeClass('notAgreed');
-              var finalDest = escape(siteURL + "/challenge-details/terms/" + challengeId + "?challenge-type=" + challengeType + "&cb=" + Math.random());
+              var finalDest = escape(tcconfig.mainURL + "/challenge-details/terms/" + challengeId + "?challenge-type=" + challengeType + "&cb=" + Math.random());
```
Comment (vikasrohit) [inline src/js/terms.js]: This change is to allow local reproduction of the issue in the vagrant env.

File: src/js/terms.js [hunk 2]
```diff
@@ -49,13 +49,13 @@ appChallengeTerms = {
...
               $.ajax({
                 url: tcconfig.apiURL + '/terms/docusign/viewURL',
                 type: 'POST',
                 data: {
                   templateId: data["docusignTemplateId"],
-                  returnUrl: siteURL + "/iframe-break/?dest=" + finalDest
+                  returnUrl: tcconfig.mainURL + "/iframe-break/?dest=" + finalDest
```
Comment (vikasrohit) [inline src/js/terms.js]: This change is to allow local reproduction of the issue in the vagrant env.

File: src/js/terms.js [hunk 3]
```diff
@@ -108,7 +108,8 @@ appChallengeTerms = {
                   'Authorization': 'Bearer ' + tcjwt.replace(/["]/g, "")
                 },
                 success: function(data) {
-                  window.location = siteURL + "/challenge-details/terms/" + challengeId + "?challenge-type=" + challengeType;
+                  console.log(tcconfig.mainURL);
+                  window.location = tcconfig.mainURL + "/challenge-details/terms/" + challengeId + "?challenge-type=" + challengeType;
```
Comment (vikasrohit) [inline src/js/terms.js]: This change is to allow local reproduction of the issue in the vagrant env.

File: src/js/terms.js [hunk 4]
```diff
@@ -144,7 +145,7 @@ appChallengeTerms = {
                   allAgreed = false;
                 }
                 var $tr = $("<tr>", {class: i % 2 == 1 ? "alt" : ""});
-                var $td1 = $("<td>").text(terms[i]["title"]).append(" (").append($("<a>", {href: siteURL + "/challenge-details/terms/detail/" + terms[i]["termsOfUseId"] + "?contestID=" + challengeId + "&challenge-type=" + challengeType}).text(agreed ? "view" : "view and agree")).append(")");
+                var $td1 = $("<td>").text(terms[i]["title"]).append(" (").append($("<a>", {href: tcconfig.mainURL + "/challenge-details/terms/detail/" + terms[i]["termsOfUseId"] + "?contestID=" + challengeId + "&challenge-type=" + challengeType}).text(agreed ? "view" : "view and agree")).append(")");
```
Comment (vikasrohit) [inline src/js/terms.js]: This change is to allow local reproduction of the issue in the vagrant env.

File: src/js/terms.js [hunk 5]
```diff
@@ -162,7 +162,18 @@ appChallengeTerms = {
             $('.loading').hide();
 
             $(".termsBtnRegister").click(function () {
-              window.location = siteURL + "/challenge-details/register/" + challengeId  + "?type=" + challengeType + "&nocache=true";
```
Comment (vikasrohit) [inline src/js/terms.js]: following change set is from community.

**Comment by parthshah** [comment]
lgtm. merging.

**Pull request merged**